docker/Dockerfile

```diff
@@ -1,34 +1,105 @@
-ARG BASE_IMAGE=python:3.8-slim-bullseye
+##
+#  Generic dockerfile for dbt image building.  
+#  See README for operational details
+##
 
-FROM $BASE_IMAGE
-ARG BASE_REQUIREMENTS_SRC_PATH
-ARG WHEEL_REQUIREMENTS_SRC_PATH
-ARG DIST_PATH
+##
+# base image (abstract)
+##
+FROM python:3.9-slim-bullseye as base
 
+# Build args
+# TODO: Can we automate this so it doesn't need to be updated with each major rev?
+# N.B. dbt-postgres is currently found in the core codebase so a value of dbt-core@1.0.latest is correct
```

Comment:
When we bump the ~major~ minor version from `1.0` to `1.1`, we'll need to update `1.0.latest` to `1.1.latest`.

Is there any way we can automate this, via [bumpversion config](https://github.com/dbt-labs/dbt-core/blob/main/.bumpversion.cfg) or [`version-bump` GitHub Action](https://github.com/dbt-labs/dbt-core/blob/main/.github/workflows/version-bump.yml)?

Reply:
I haven't found an easy way to do that yet (hence the todo).  But I can put some more time into it today.

Reply:
I played around with a few different options and just adding the `Dockerfile` to `.bumpversion.cfg` seems to be the best solution that isn't terribly janky.  I think this is fine, although it does mean that the definition of what's considered "up to date" will only be as accurate as the last time you sourced the `Dockerfile`. 

I'd really prefer that it dynamically set that at build time, but Docker doesn't really give you much in the way of tools for that sort of thing outside of external scripting, which is also something I'd rather avoid.

Does this work for you @jtcohen6 ?

Reply:
Including Dockerfile in bumpversion makes a lot of sense to me. Were you able to confirm that the bumpversion step captured the branch upgrade above? Or would we need to switch this from a branch specification to a specific version tag?

Reply:
I actually had to switch to a tag based selection process as I couldn't get bumpversion to recognize `latest` as a branch.  AFAICT this should work equally well and if someone wants to manually specify the ref they can still use anything supported by pip.